### PR TITLE
Update folder structure and Use ballerina-platform/plugin-gradle

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Purpose
Change the folder structure  similar to other standard library structure and gradle files updated to uses ballerina gradle plugin

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (ThisaruGuruge) [inline build.gradle]: Let's keep the original year.

Comment (MohamedSabthar) [reply build.gradle]: Fixed.

File: build.gradle [hunk 2]
```diff
@@ -16,58 +16,81 @@
...
+    tagTemplate = 'v$version'
+
+    git {
+        requireBranch = "release-${moduleVersion}"
+        pushToRemote = 'origin'
     }
 }
+
+task build {
+    dependsOn('serdes-ballerina:build')
```
Comment (ThisaruGuruge) [inline build.gradle:96]:
```suggestion
    dependsOn('serdes-ballerina:build')
    dependsOn('serdes-native:build')
```

File: ballerina/tests/unit_tests.bal [hunk 3]
```diff
@@ -282,8 +282,8 @@ type Member record {
 
 @test:Config{}
 public isolated function testRecordWithNil() returns error? {
-    Contact phone1 = {mobile: "+123456", home: "789"};
-    Member member1 = {name: "foo", salary: 1.23, contact: phone1};
+    // Contact phone1 = {mobile: "+123456", home: "789"};
+    // Member member1 = {name: "foo", salary: 1.23, contact: phone1};
```
Comment (ThisaruGuruge) [inline ballerina/tests/unit_tests.bal]: Why do we commenting these lines? (If these are no more needed, let's remove them).

Comment (MohamedSabthar) [reply ballerina/tests/unit_tests.bal]: These lines are not used in assert statements hence removed now.

File: build.gradle (hunk 2 above)
Comment (MohamedSabthar) [reply build.gradle:96]: Suggestion commited.

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**